Library/Homebrew/extend/os/linux/requirements/osxfuse_requirement.rb

```diff
@@ -6,6 +6,11 @@
 class OsxfuseRequirement < Requirement
   extend T::Sig
 
+  def initialize(tags = [])
+    odeprecated "depends_on :osxfuse", 'depends_on "libfuse"'
```

Comment:
@Homebrew/linux how would you like this handled in formulae? 

```ruby
on_linux do
  depends_on "libfuse"
end
```

Reply:
Yep, that looks good to me! @iMichka ?

Reply:
Yes @sjackman and @MikeMcQuaid, this is how it should be written.
LGTM

Reply:
Maybe:
`odeprecated "depends_on :osxfuse", 'depends_on "libfuse for linux only"'` ?

Reply:
The wanted syntax:
```
on_linux do
  depends_on "libfuse"
end
```

Reply:
Thanks @iMichka, @sjackman, @jonchang!